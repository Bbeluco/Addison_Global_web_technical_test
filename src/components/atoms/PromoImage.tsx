import React from 'react';
import {Image, View} from 'react-native';
import {PromoImageType} from '../../types/promoImageType';

export function PromoImage({
  imageLink,
  styleChildren,
}: PromoImageType): React.JSX.Element {
  return (
    <View>
      <Image style={styleChildren} source={{uri: imageLink}} />
    </View>
  );
}
