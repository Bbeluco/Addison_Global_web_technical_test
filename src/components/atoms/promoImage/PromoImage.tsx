import React from 'react';
import {Image, View} from 'react-native';
import {PromoImageType} from '@customTypes/promoImageType';

export function PromoImage({
  imageLink,
  styleChildren,
}: PromoImageType): React.JSX.Element {
  return (
    <View>
      <Image
        style={styleChildren}
        source={{uri: imageLink}}
        testID={'promo_image'}
      />
    </View>
  );
}
