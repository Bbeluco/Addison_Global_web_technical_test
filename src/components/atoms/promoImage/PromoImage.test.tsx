import React from 'react';
import {render, screen} from '@testing-library/react-native';
import {API_RESULT_SUCSESS_MOCK} from '@constants';
import {PromoImage} from './PromoImage';

describe('<PromoImage />', () => {
  it('should render and has sorce passed as prop', () => {
    const src = API_RESULT_SUCSESS_MOCK[0].heroImageUrl;
    render(<PromoImage styleChildren={{}} imageLink={src} />);
    const image = screen.getByTestId('promo_image');
    expect(image.props.source).toEqual({uri: src});
    expect(screen.toJSON()).toMatchSnapshot();
  });
});
