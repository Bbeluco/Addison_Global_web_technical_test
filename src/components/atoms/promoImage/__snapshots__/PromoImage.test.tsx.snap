// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PromoImage /> should render and has sorce passed as prop 1`] = `
<View>
  <Image
    source={
      {
        "uri": "https://via.placeholder.com/600x300",
      }
    }
    style={{}}
    testID="promo_image"
  />
</View>
`;
